qcodes/dataset/measurements.py

```diff
@@ -669,8 +478,7 @@ def __enter__(self) -> DataSaver:
         self.datasaver = DataSaver(
                             dataset=self.ds,
                             write_period=self.write_period,
-                            interdeps=self._interdependencies,
-                            write_in_background=self._write_in_background)
```

Comment:
This is tecnially an API break but I don't think anyone is inializing the datasaver directly

Reply:
exactly. as per our docs, inititlzng DataSaver is NOT a pulbic API :)